ServerlessLibraryAPI/ClientApp/src/App.js

```diff
@@ -1,42 +1,53 @@
-import React, { Component } from 'react';
-import { Switch, Route } from 'react-router-dom';
-
-import './App.css';
-import { Header } from './components/Header';
-import { Login } from './components/Login';
+import React, { Component } from "react";
+import { Switch, Route, withRouter } from "react-router-dom";
+import { connect } from "react-redux";
+import Main from "./components/Main/Main";
+import "./App.css";
+import { Header } from "./components/Header";
+import DetailView from "./components/DetailView/DetailView";
+import { samplesReceived } from "./actions/FilterChangeActions";
+import { Login } from "./components/Login";
 
 class App extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-        samples: []
-    }
-    this.getCurrentSample = this.getCurrentSample.bind(this)
-}
-
-getCurrentSample(id)
-{
-  return this.state.samples.filter(s=>s.title===id);
-}
-
+  componentDidMount() {
+    fetch("https://www.serverlesslibrary.net/api/Library")
```

Comment:
can you please move this code into a separate service, eg libraryService ?
getAll and getById (which is being used in detail view) can go into the service